openedx/core/djangoapps/api_admin/api/v1/views.py

```diff
@@ -0,0 +1,83 @@
+"""
+API Views.
+"""
+
+from django.contrib.auth.models import User
+from django.utils.translation import ugettext_lazy as _
+
+from edx_rest_framework_extensions.authentication import JwtAuthentication
+from rest_framework.authentication import SessionAuthentication
+from rest_framework.exceptions import NotFound, PermissionDenied
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.generics import ListAPIView
+from rest_framework_oauth.authentication import OAuth2Authentication
+
+from openedx.core.djangoapps.api_admin.api.v1 import serializers as api_access_serializers
+from openedx.core.djangoapps.api_admin.models import ApiAccessRequest
+
+
+class ApiAccessRequestView(ListAPIView):
+    """
+    Return `API Access Requests` in the form of a paginated list.
+
+    Raises:
+        NotFound: Raised if user with `username` provided in `GET` parameters does not exist.
+        PermissionDenied: Raised if `username` is provided in `GET` parameters but the requesting
+            user does not have access rights to filter results.
+
+    Example:
+        `GET: /api-admin/api/v1/api_access_request/`
+        {
+            "count": 1,
+            "num_pages": 1,
+            "current_page": 1,
+            "results": [
+                {
+                    "id": 1,
+                    "created": "2017-09-25T08:41:48.934364Z",
+                    "modified": "2017-09-25T08:42:04.185209Z",
+                    "user": 6,
+                    "status": "denied",
+                    "website": "https://www.example.com/",
+                    "reason": "Example",
+                    "company_name": "Example Name",
+                    "company_address": "Silicon Valley",
+                    "site": 1,
+                    "contacted": true
+                }
+            ],
+            "next": null,
+            "start": 0,
+            "previous": null
+        }
+    """
+    authentication_classes = (JwtAuthentication, OAuth2Authentication, SessionAuthentication,)
+    permission_classes = (IsAuthenticated, )
+    serializer_class = api_access_serializers.ApiAccessRequestSerializer
+
+    def get_queryset(self):
+        username = self.request.query_params.get('username', None)
+        queryset = ApiAccessRequest.objects.all()
+
+        # Filter based on given username, if requesting user has staff access.
+        if self.request.user.is_staff:
+            if username:
```

Comment:
Just a question, Can we eliminate that check if we have the empty string as the default value instead of `None`?

Reply:
@asadiqbal08 this will check both cases of `username` being empty (`''`) or non present (`None`). We only want to filter when `username` is present and is not empty.